fv3core/stencils/updatedzd.py

```diff
@@ -151,154 +256,144 @@ def out(zs: sd, zh: sd, ws: sd, dt: float):
+    zh: FloatField,
+    crx: FloatField,
+    cry: FloatField,
+    xfx: FloatField,
+    yfx: FloatField,
+    wsd: FloatField,
+    dt: float,
+):
     grid = spec.grid
-
-    ndif[-1] = ndif[-2]
-    damp_vtd[-1] = damp_vtd[-2]
+    halo = grid.halo
 
     crx_adv = utils.make_storage_from_shape(
-        crx.shape, grid.compute_origin(add=(0, -grid.halo, 0))
+        crx.shape, grid.compute_origin(add=(0, -halo, 0))
     )
     cry_adv = utils.make_storage_from_shape(
-        cry.shape, grid.compute_origin(add=(-grid.halo, 0, 0))
+        cry.shape, grid.compute_origin(add=(-halo, 0, 0))
     )
     xfx_adv = utils.make_storage_from_shape(
-        xfx.shape, grid.compute_origin(add=(0, -grid.halo, 0))
+        xfx.shape, grid.compute_origin(add=(0, -halo, 0))
     )
     yfx_adv = utils.make_storage_from_shape(
-        yfx.shape, grid.compute_origin(add=(-grid.halo, 0, 0))
+        yfx.shape, grid.compute_origin(add=(-halo, 0, 0))
     )
     ra_x = utils.make_storage_from_shape(
-        crx.shape, grid.compute_origin(add=(0, -grid.halo, 0))
+        crx.shape, grid.compute_origin(add=(0, -halo, 0))
     )
     ra_y = utils.make_storage_from_shape(
-        cry.shape, grid.compute_origin(add=(-grid.halo, 0, 0))
+        cry.shape, grid.compute_origin(add=(-halo, 0, 0))
     )
 
-    gam = utils.make_storage_from_shape(zs.shape, grid.full_origin())
-    edge_profile(
+    edge_profile_stencil(
         crx,
         xfx,
         crx_adv,
         xfx_adv,
-        dp0,
-        gam,
-        origin=(grid.is_, grid.jsd, 0),
-        domain=(grid.nic + 1, grid.njd, grid.npz + 1),
-    )
-    # edge_python(crx, xfx, crx_adv, xfx_adv, dp0, gam, slice(grid.is_, grid.ie
-    # + 2), slice(grid.jsd, grid.jed+1),  qe1_2, gam_2)
-
-    gam = utils.make_storage_from_shape(zs.shape, grid.full_origin())
-    edge_profile(
         cry,
         yfx,
         cry_adv,
         yfx_adv,
         dp0,
-        gam,
-        origin=(grid.isd, grid.js, 0),
-        domain=(grid.nid, grid.njc + 1, grid.npz + 1),
+        origin=grid.compute_origin(add=(-halo, -halo, 0)),
+        domain=grid.domain_shape_compute(add=(2 * halo, 2 * halo, 1)),
     )
-
     ra_stencil_update(
         grid.area,
         xfx_adv,
-        ra_x,
         yfx_adv,
+        ra_x,
         ra_y,
-        origin=grid.compute_origin(add=(-grid.halo, -grid.halo, 0)),
-        domain=(grid.nid, grid.njd, grid.npz + 1),
+        origin=grid.compute_origin(add=(-halo, -halo, 0)),
+        domain=grid.domain_shape_compute(add=(2 * halo, 2 * halo, 1)),
     )
 
-    # TODO, when consoldiating k plitting, change this too
-    data = {}
-    for varname in ["zh", "crx_adv", "cry_adv", "xfx_adv", "yfx_adv", "ra_x", "ra_y"]:
-        data[varname] = locals()[varname]
+    ndif[-1] = ndif[-2]
+    damp_vtd[-1] = damp_vtd[-2]
+    kstarts = utils.get_kstarts({"ndif": ndif, "damp": damp_vtd}, grid.npz + 1)
 
-    col = {"ndif": ndif, "damp": damp_vtd}
+    for ki, nk in kstarts:
```

Comment:
ah! This is a good step forward. For future refactors getting rid of k_split_run, I think we might just want to hard code the first 2-3 'intervals' inside the stencils, but this is a good first step that keeps it more general, and we should probably discuss before hard coding in those levels. Also probably can't do that right now as this is calling other stencil methods